### PR TITLE
[302] Parse first GroupSchema SchemaType before trying DataType

### PR DESCRIPTION
**Pull request by JanC**
This PR changes the generation of composed types (oneOf, anyOf) to prefer the composed type instead of the `object` when both are present in the schema

See ticket for detailed description

https://github.com/yonaskolb/SwagGen/issues/302

### CODE REVIEW

**Comment by yonaskolb** [comment]
Thanks @JanC. Could you also add an example to the test spec here https://github.com/yonaskolb/SwagGen/blob/master/Specs/TestSpec/spec.yml and also a changelog entry

**Comment by JanC** [comment]
somehow the CI keeps on failing but I think it's unrelated to this PR.

@yonaskolb any idea why?

File: Specs/TestSpec/spec.yml [hunk 1]
```diff
@@ -358,6 +358,15 @@ components:
           woof: "#/components/schemas/Dog"
           dog: "#/components/schemas/Dog"
           cat: "#/components/schemas/Cat"
+    SingleAnimalWithObjectType:
```
Comment (yonaskolb) [inline Specs/TestSpec/spec.yml:361]: Isn’t this the same as the above SingleAnimal?

Comment (JanC) [reply Specs/TestSpec/spec.yml:361]: except it has the additional `type: object`

I wanted to make sure there is a test for both cases: with the `type: object` and without (the SingleAnimal)

Comment (yonaskolb) [reply Specs/TestSpec/spec.yml:361]: Ah I see you added it now 👍

**Review by yonaskolb — APPROVED** [review]

**Comment by yonaskolb** [comment]
Thanks @JanC! Yeah the CI is failing for a different reason, and I haven't had time to look into it. Not sure what's happening

**Comment by JanC** [comment]
thanks for merging 🙏

**Pull request merged**